system/draft/test_upgrade_docker_7.py

```diff
@@ -7,92 +7,177 @@
 from datetime import datetime, timedelta, timezone
 import json
 import testinfra
+from system.docker_setup import client, pool_builder, pool_starter,\
+    DOCKER_BUILD_CTX_PATH, DOCKER_IMAGE_NAME, NODE_NAME_BASE, NETWORK_NAME
 
 
 # TODO dynamic install of old version to upgrade from
 # TODO INDY-2132 and INDY-2125
 @pytest.mark.asyncio
-async def test_pool_upgrade_positive(docker_setup_and_teardown):
-    await pool.set_protocol_version(2)
+async def test_pool_upgrade_positive(
+        docker_setup_and_teardown, payment_init, initial_token_minting, pool_handler, wallet_handler,
+        check_no_failures_fixture
+):
+    # ------------------EXTRA NODE SETUP--------------------------------------------------------------------------------
+    # create extra node
+    new_node = pool_starter(
+        pool_builder(
+            DOCKER_BUILD_CTX_PATH, DOCKER_IMAGE_NAME, 'new_node', NETWORK_NAME, 1
+        )
+    )[0]
+
+    GENESIS_PATH = '/var/lib/indy/sandbox/'
+
+    # put both genesis files
+    print(new_node.exec_run(['mkdir', GENESIS_PATH], user='indy'))
+
+    for _, prefix in enumerate(['pool', 'domain']):
+        bits, stat = client.containers.get('node1'). \
+            get_archive('{}{}_transactions_genesis'.format(GENESIS_PATH, prefix))
+        assert new_node.put_archive(GENESIS_PATH, bits)
+
+    new_ip = '10.0.0.9'
+    PORT_1 = '9701'
+    PORT_2 = '9702'
+    new_alias = 'Node8'
+
+    # initialize
+    assert new_node.exec_run(
+        ['init_indy_node', new_alias, new_ip, PORT_1, new_ip, PORT_2, '000000000000000000000000000node8'],
+        user='indy'
+    ).exit_code == 0
+
+    # upgrade it to the target version of pool upgrade command
+    plenum_ver = '1.11.0'
+    plenum_pkg = 'indy-plenum'
+    node_ver = '1.11.0'
+    node_pkg = 'indy-node'
+    sovrin_ver = '1.1.60'
+    sovrin_pkg = 'sovrin'
+    plugin_ver = '1.0.4'
+    assert new_node.exec_run(
+        ['apt', 'update'],
+        user='root'
+    ).exit_code == 0
+    assert new_node.exec_run(
+        ['apt', 'install',
+         '{}={}'.format(sovrin_pkg, sovrin_ver),
+         '{}={}'.format(node_pkg, node_ver),
+         '{}={}'.format(plenum_pkg, plenum_ver),
+         '{}={}'.format('sovtoken', plugin_ver),
+         '{}={}'.format('sovtokenfees', plugin_ver),
+         '-y'],
+        user='root'
+    ).exit_code == 0
+
+    # start
+    assert new_node.exec_run(
+        ['systemctl', 'start', 'indy-node'],
+        user='root'
+    ).exit_code == 0
```

Comment:
Actually I'd prefer to start this node _after_ transactions are written and upgrade is performed